docs/adr/0008-split-bindings-to-modules-automatically.md

```diff
@@ -0,0 +1,64 @@
+# [short title of solved problem and solution]
+
+* Status: [proposed superseded by [gogo](0008-split-bindings-to-modules-automatically.md)]
+* Deciders: []
+* Date: [2022-07-11 when the decision was last updated]
+
+## Context and Problem Statement
+
+In his current state, uniffi only allow to import any generated code to a single file. This is embarassing because in languages used to make bindings, the user has to import submodules.
+
+## Considered Options
+
+* [option 1] : Force the user to rewrite all import in an __init__ for each python file. Mention it on the documentation. Not recommended.
+* [option 2] : Fork Uniffi to write two new types of objects know as directory and file. EG:
+
+directory house {
+    file small_house {
+        interface SmallHouse {
+            constructor();
+            [Name=build_house]
+            constructor();
+            string get_house_street();
+        };
+    }
+}
+
+ Recommended.
```

Comment:
There's a 3rd option here: Describe the module structure in `uniffi.toml`, which would be my first instinct to use.  The UDL file is shared between all bindings languages, so it's best used for things that can be generalized across all languages.  The `uniffi.toml` file has a section for each bindings languages, so it's best used for things specific to a language.

I'm pretty sure packages/modules are going to need to be described per-language.  The example you provided seems very python-specific to me.

Reply:
Good point. I am going to mention the way with several UDL files in order to have several python/kotlin modules. It is less pythonic.